examples/ansible/code/Synchronize_DNS_Zones

```diff
@@ -0,0 +1,50 @@
+- name: Sync DNS
+  hosts: <your Bind-Server>
+  tasks:
+    - name: get Managed Zones Names
+      ansible.builtin.set_fact:
+        zone_names: "{{ query('netbox.netbox.nb_lookup', 'zones', plugin='netbox_dns',
+                           api_endpoint='https://<nextcloud-url>/',
+                           api_filter='status=active status=deprecated',
+                           token='<redacted>')
+                    | map(attribute='value') | map(attribute='name') }}"
+
+    - name: Generate Zone Files
+      template:
+          src: dns.zone.j2
+          dest: "/path-to-your-zone-folder/{{ item }}.zone"
+      vars:
+          zonename: "{{ item }}"
+          zone: "{{ query('netbox.netbox.nb_lookup', 'zones', plugin='netbox_dns',
+                           api_endpoint='https://<nextcloud-url>/',
+                           api_filter='name='+zonename,
+                           token='<redacted>')
+                    | map(attribute='value')
+                    | first }}"
+          records: "{{ query('netbox.netbox.nb_lookup', 'records', plugin='netbox_dns',
+                             api_endpoint='https://<nextcloud-url>/',
+                             api_filter='zone='+zone.name,
+                             token='<redacted>')
+                       | map(attribute='value') }}"
+      loop: "{{ zone_names }}"
+      notify: restart bind9
+
+    - name: Collect Directory contents
+      shell: >
+        ls -1 /path-to-your-zone-folder/ | sed 's/\.[^.]*$//'
+      register: files
+      changed_when: False
+
+    - name: Remove unwanted Files
+      file:
+        path: "/path-to-your-zone-folder/{{ item }}.zone"
+        state: absent
+      with_items: "{{ files.stdout_lines }}"
+      when: >
+        item not in zone_names
+
+  handlers:
+   - name: restart bind9
+     ansible.builtin.service:
+       name: bind9
+       state: restarted
```

Comment:
Currently your playbook does not support views. Could you please add them, at least optionally? In NetBox DNS 1.0.0, views are no longer optional (mainly for internal reasons, there is a default view that is largely equivalent to having no view at all), but if there is more than one view your playbook will overwrite zones files with the same name.

Reply:
Im not quite sure how to implement it, because atm my setup doesnt use views (only the default view). Would it be sufficient to add a variable with a list with active_views for the server and only get the zones that are in that list and push them to the bind server? For a better overview i would consider grouping them into folders per view.

Reply:
The easiest way (after all, it's just an example) you could include a sub-directory named after the view in which to put the zone files. It's simple, and it gets rid of the problem overwriting zone files.

Reply:
Yeah, that was also on my mind.